src/tanks/Tancik.java

```diff
@@ -9,13 +9,21 @@
     public class Tancik extends AdvancedRobot{
 
         public Vector<Gene> genes;
+        public Vector<Gene> genesRun;
+        public Vector<Gene> genesOnScanned;
+        public Vector<Gene> genesOnHit;
         public Tancik(Vector<Gene> genes) {
+
+            this.genesRun = genes;
+            this.genesOnScanned = genes;
+            this.genesOnHit = genes;
```

Comment:
Radek 17-19 nedava smysl, protoze vytvorene skupiny genu, ktere maji provest danou akci jsou pokazde naplneny stejnym obsahem (genes).

V tomto pripade je dobre udelat pro kazdou metodu zvlast cyklus `for` (v tomto pripade to budou tri). Kazdy cyklus bude cyklit jen do urciteho procenta poctu genu (definovano v konfigu). Tyto geny se priradi do sve hodnoty (napr. `this.genesRun`) a zaroven se prirazena hodnota smaze z vektoru `Genes`, aby uz nemohla byt prirazena pro jinou akci (napr. `genesOnScanned`)

Pseudokod mych myslenek:
```java
public static double percRun = 0.2;  // tato hodnota uz je definovana v Config.java
for (int gene; gene < percRun*genes) {
genesRun.add(gene);
// + kod na vymazani gene z vektoru genes. Neco jako genes.pollFirst(gene) - nejsem si jista, jestli funkce pollFirst dela to, co potrebujeme, je treba to zjistit.
}
```

Reply:
Tak jak píše Šárka ... je potřeba projít celý vektor a nasázet pouze geny do metod v jednotlivých intervalech od 0 do i, od i do j, atd.